crypto/provider_core.c

```diff
@@ -203,8 +203,8 @@ int ossl_provider_disable_fallback_loading(OSSL_LIB_CTX *libctx)
 {
     struct provider_store_st *store;
 
-    if ((store = get_provider_store(libctx)) != NULL) {
-        CRYPTO_THREAD_write_lock(store->lock);
+    if ((store = get_provider_store(libctx)) != NULL
+            && CRYPTO_THREAD_write_lock(store->lock)) {
```

Comment:
I think this should be:

``` C
if ((store = get_provider_store(libctx)) != NULL) {
    if (!CRYPTO_THREAD_write_lock(store->lock))
        return 0;
    ...
}
```

The call has failed to do what it was meant to, so it ought to return an error.

Reply:
the two fragments are equivalent, but your suggestion is more clear so I pushed an updated fixup.